oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResultRowAsyncIterator.java

```diff
@@ -253,9 +252,8 @@ class ElasticQueryScanner {
                     }
             );
 
-            if (LOG.isTraceEnabled()) {
-                LOG.trace("Kicking initial search for query {}", ElasticIndexUtils.toString(searchReq));
-            }
+
+            LOG.trace("Kicking initial search for query {}", searchReq);
```

Comment:
I think we should keep these if conditions, as we will rarely enable these trace logs and parameter evaluation will happen for almost everything though it is not being logged.
https://logging.apache.org/log4j/1.2/manual.html#performance

Reply:
that's not needed anymore. Before this change, we had to create the `String` representation because the Elasticsearch client did not return it. The behavior has been fixed with the latest client (updated before this PR). I have removed the if-block because we pass the SearchRequest object reference now. Log4j does not call the toString except when  the log level is >= TRACE